### ISSUE TITLE
mp4dash throws ERROR: float division by zero 

### ISSUE THREAD

**Issue by calvincorreli** [axiomatic-systems/Bento4#243]
I get this fairly regularly.

People upload videos, which I process with ffmpeg, then mp4fragment, and then mp4dash.

My command line is something like this, but the parameters don't seem to matter:

```mp4dash --verbose --profiles=on-demand --no-split --hls -o hls1 ffmpeg-output-general-fragmented.mp4```

Output:

```Profiles: urn:mpeg:dash:profile:isoff-on-demand:2011
Parsing media file 1: ffmpeg-output-general-fragmented.mp4
ERROR: float division by zero
```

Here's the output from mp4info:

```$ mp4info ffmpeg-output-general-fragmented.mp4

File:
  major brand:      isom
  minor version:    200
  compatible brand: isom
  compatible brand: iso2
  compatible brand: avc1
  compatible brand: iso6
  compatible brand: mp41
  compatible brand: iso5
  fast start:       yes

Movie:
  duration:   0 ms
  time scale: 1000
  fragments:  yes

Found 3 Tracks
Track 1:
  flags:        7 ENABLED IN-MOVIE IN-PREVIEW
  id:           1
  type:         Video
  duration: 0 ms
  language: und
  media:
    sample count: 0
    timescale:    30000
    duration:     0 (media timescale units)
    duration:     0 (ms)
    bitrate (computed): 236.567 Kbps
    sample count with fragments: 10398
    duration with fragments:     10408398
    duration with fragments:     346947 (ms)
  display width:  1280.000000
  display height: 720.000000
  Sample Description 0
    Coding:      avc1 (H.264)
    Width:       1280
    Height:      720
    Depth:       24
    AVC Profile:          100 (High)
    AVC Profile Compat:   0
    AVC Level:            42
    AVC NALU Length Size: 4
    AVC SPS: [6764002aacd9405005bb011000003e90000ea600f1831960]
    AVC PPS: [68ebe3cb22c0]
    Codecs String: avc1.64002A
Track 2:
  flags:        7 ENABLED IN-MOVIE IN-PREVIEW
  id:           2
  type:         Audio
  duration: 0 ms
  language: dan
  media:
    sample count: 0
    timescale:    44100
    duration:     0 (media timescale units)
    duration:     0 (ms)
    bitrate (computed): 96.007 Kbps
    sample count with fragments: 14944
    duration with fragments:     15303289
    duration with fragments:     347013 (ms)
  Sample Description 0
    Coding:      mp4a (MPEG-4 Audio)
    Stream Type: Audio
    Object Type: MPEG-4 Audio
    Max Bitrate: 96000
    Avg Bitrate: 0
    Buffer Size: 0
    Codecs String: mp4a.40.2
    MPEG-4 Audio Object Type: 2 (AAC Low Complexity)
    MPEG-4 Audio Decoder Config:
      Sampling Frequency: 44100
      Channels: 2
    Sample Rate: 44100
    Sample Size: 16
    Channels:    2
Track 3:
  flags:        7 ENABLED IN-MOVIE IN-PREVIEW
  id:           3
  type:         Unknown [tmcd]
  duration: 0 ms
  language: ```
  media:
    sample count: 0
    timescale:    30000
    duration:     0 (media timescale units)
    duration:     0 (ms)
    bitrate (computed): 0.000 Kbps
    sample count with fragments: 1
    duration with fragments:     0
    duration with fragments:     0 (ms)
  Sample Description 0
    Coding:      tmcd
```

I'm guessing it's because of Track 3 with a duration of 0, but I'm not sure, as there's no more information other than the above.

I've tried to figure out how to remove this track using ffmpeg, but so far I've been unsuccessful. I've tried ``-dn``, ``-sn``, ``-map 0:0 -map 0:1`` and other things that seem like they should work, but the third track (third rail? haha) remains, and the division by zero remains as well.

**Comment by calvincorreli** [comment]
After some more digging, I found that my ffmpeg had ``-movflags frag_keyframe+empty_moov+faststart``.

When I remove the ``empty_moov`` part, the division by zero disappears. So that's fantastic. 

I'm not entirely sure why it was there in the first place. I seemed to have found in my research that I should use ffmpeg to fragment, even though I call mp4fragment later, because ... something? I forget.
